Objects/tupleobject.c

```diff
@@ -314,15 +316,22 @@ tuple_repr(PyObject *self)
 #define _PyHASH_XXROTATE(x) ((x << 13) | (x >> 19))  /* Rotate left 13 bits */
 #endif
 
-/* Tests have shown that it's not worth to cache the hash value, see
-   https://bugs.python.org/issue9685 */
 static Py_hash_t
 tuple_hash(PyObject *op)
 {
     PyTupleObject *v = _PyTuple_CAST(op);
+
+    // For the empty singleton, we don't need to dereference the pointer
+    if (op == (PyObject *)&_Py_SINGLETON(tuple_empty)) {
```

Comment:
You could set the hash of `()` statically in `pycore_runtime_init.h`, then this test would be unnecessary.

Reply:
Yes, but interestingly, this is measurably faster -- it doesn't have to chase the pointer.

Reply:
Seems like the difference here is in the noise, so might as well do this at build time and skip this extra branch.